src/matlab2tikz.m

```diff
@@ -6244,3 +6202,89 @@ function checkDeprecatedEnvironment(minimalVersions)
     end
 end
 % ==============================================================================
+
+% DEPRECATED ===================================================================
```

Comment:
Yeah, we sort of need to decide a deprecation policy I guess. 
For big projects, simply deleting and looking it up is more time consuming than leaving it in for a few releases. 

I would leave it throughout the next release and delete at the second next release.